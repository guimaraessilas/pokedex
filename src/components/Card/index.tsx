import React from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Pokemon } from '../../types/Pokemon';
import { colors } from '../../utils/colors';
import PokemonTypeList from '../PokemonTypeList';

class Card extends React.PureComponent {
  render() {
    const { name, types, id, onPress } = this.props as Pokemon & {
      onPress: () => void;
    };

    return (
      <TouchableOpacity
        onPress={onPress}
        style={[
          styles.container,
          { backgroundColor: `${colors[types[0].type.name]}cc` },
        ]}
      >
        <Text style={styles.title}>{name}</Text>
        <View style={{ flexDirection: 'row', width: '100%' }}>
          <PokemonTypeList types={types} />
          <Image
            source={{
              uri: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`,
            }}
            style={styles.picture}
          />
        </View>
      </TouchableOpacity>
    );
  }
}

export default Card;

const styles = StyleSheet.create({
  picture: {
    height: 150,
    width: 150,
    marginLeft: -10,
    marginTop: -20,
  },
  container: {
    width: '45%',
    margin: 10,
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: 12,
  },
  title: {
    fontSize: 24,
    textTransform: 'capitalize',
    alignSelf: 'flex-start',
    color: 'white',
    fontWeight: '700',
    marginBottom: 8,
    marginTop: 6,
  },
});
