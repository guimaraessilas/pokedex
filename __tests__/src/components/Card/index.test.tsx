import React from 'react';
import 'react-native';
import { create } from 'react-test-renderer';
import Card from '../../../../src/components/Card';

jest.mock('@expo/vector-icons', () => 'Icon');

const props = {
  name: 'bulbasaur',
  types: [{ type: { name: 'grass' } }, { type: { name: 'poison' } }],
  onPress: () => jest.fn(),
  sprites: {
    front_default: '',
  },
  captured: false,
};

const setup = () => create(<Card {...props} />).toJSON();

it('Should render Card with icon', () => {
  const tree = setup();
  expect(tree).toMatchSnapshot();
});
