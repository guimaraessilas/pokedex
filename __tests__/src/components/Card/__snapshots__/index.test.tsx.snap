// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Card with icon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#7AC74Ccc",
      "borderRadius": 20,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "width": "45%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 22,
          "fontWeight": "700",
          "marginBottom": 8,
          "marginTop": 6,
          "textTransform": "capitalize",
        }
      }
    >
      bulbasaur
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginLeft": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "alignSelf": "flex-start",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 100,
              "flexDirection": "column",
              "height": 32,
              "justifyContent": "center",
              "marginBottom": 4,
              "marginRight": 12,
              "paddingHorizontal": 5,
            },
            {
              "backgroundColor": "#7AC74C",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
              "padding": 5,
              "textTransform": "capitalize",
            }
          }
        >
          grass
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 100,
              "flexDirection": "column",
              "height": 32,
              "justifyContent": "center",
              "marginBottom": 4,
              "marginRight": 12,
              "paddingHorizontal": 5,
            },
            {
              "backgroundColor": "#A33EA1",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
              "padding": 5,
              "textTransform": "capitalize",
            }
          }
        >
          poison
        </Text>
      </View>
    </View>
    <Image
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "height": 150,
          "marginLeft": -10,
          "marginTop": -20,
          "width": 150,
        }
      }
    />
  </View>
</View>
`;
